### ISSUE TITLE
Meta Discussion of Extension Ownership

### ISSUE THREAD

**Issue by JMurph2015** [vibou/vibou.gTile#81]
Hi @vibou ,
I think its become pretty noticeable that there is growing community support for your extension.  However, keeping up with PR's for these contributions (and/or personal contributions to do the work) is understandably a challenge in its own right, especially since it seems that pretty much everyone working on this are hobbyists like me.

_Would you consider making an organization to manage the repo so that other people can also review PR's?_  

Gnome and Linux in general are moving pretty fast (and breaking things) these days with the jump to Wayland, so these PR's are becoming more critical to keep the extension functioning in a minimal sense.  I personally really appreciate the work you've done and the extension you've created, but there seems to be lots of enthusiasm dying on the vine due to the PR lag time; so it may be time to take the next step in growing the project.
Thanks,
Joseph Murphy

**Comment by scherepanov** [comment]
Discussion should include ownership of gTile on extensions.gnome.org

**Comment by maweki** [comment]
We now had two gTile forks with new features at extensions.gnome.org, which I both put on hold until this issue is somehow resolved.

In general we want people to work with upstream instead of forking.

**Comment by scherepanov** [comment]
Can you put gTile master to extensions.gnome.org? It is good enough, only thing that is missing is a fix for Gnome 3.25.

**Comment by maweki** [comment]
I can't do a release. The best move would be to create an organization and move a fork there. I know that PRs and Issues get lost this way, but I am sure the community can handle it.

But first I will try to reach out to @vibou. Maybe we can do it with their approval and move the current repository.

**Comment by jwarkentin** [comment]
Something should happen. While extensions are one of the greatest features of Gnome, unmaintained extensions are one of the greatest problems. This is such a useful extension but it's hardly usable right now and I don't see any good replacements.

**Comment by scherepanov** [comment]
With much hesitance I will look into creating an organization. Please comment if you want to be a member.

**Comment by cwh1te** [comment]
Having followed this extension and its forks for a few months now, and especially in the absence of input from @vibou, I would definitely support you in your efforts, @scherepanov. I've been running your fork since June and you've done a great job of adding features and keeping it stable with Gnome updates.
I'd love to join if you create an organization - I may not be able to contribute much code but can definitely test, give input on PRs, and respond to some issues.

**Comment by jwarkentin** [comment]
I would be happy to participate as a collaborator. While we all have busy lives I'm sure that with a few collaborators working together we could keep the project moving.

**Comment by maweki** [comment]
Ok, once you've done this, decide on who is responsible for uploading to extensions.gnome.org. If they already have an account there all the better. Write to the gnome-shell mailing list referencing this ticket and my approval with the wish for ... to become the new canonical fork with ... responsible for uploading to e.g.o because of the reasons discussed here (large-ish community, unresponsive original author, stuck PRs, etc.).

Once you're done you can upload new stuff to e.g.o. Please don't switch around a thousand lines of code and submit. We do incremental reviews. Medium-sized changesets are better than huge ones.

**Comment by JMurph2015** [comment]
@scherepanov I'd also like to help out with the new org :smile: (just in case it wasn't clear before)

**Comment by scherepanov** [comment]
@maweki Organization github.com/gTile formed with 4 members, code is there ready to be posted to e.g.o, I sent email to gnome-shell-list@gnome.org with request to reassign gTile extension ownership to me.

**Comment by lexruee** [comment]
@scherepanov Why not just forking the repo to the new organization? Otherwise, the entire git history is lost when you make a new repo with the initial commit...

**Comment by scherepanov** [comment]
I do not have experience with github organizations. Have concerns that if initial repo is forked from existing repo, it will be some side effects in future.
And, frankly I do not see value in keeping history. On recent commit I doubled code size. Even if someone have it's own fork and made changes there, it have to be ported again - master changed way too much. Clean initial commit without history was done on purpose.

**Comment by maweki** [comment]
keeping the history is better because that makes it easier for people to rebase their pull-requests. It's also common courtesy and keeps a little bit of a track of copyright.

I'd implore you to at least push the original history and put your code up from there, if you're not using the fork-functionality.

**Comment by scherepanov** [comment]
Most likely you are right... I do not have a lot of experience with open source projects, and definitely keeping original history and track of copyright is some of ethical issue. Never thought about it.
gTile now supported by organization... I will put an issue about keeping history, if voted it will be there. I am intended to do all changes by majority votes after initial commit.

**Comment by lexruee** [comment]
I agree with @maweki. Kepping the history is also helpful to understand old code changes and bug fixes.

@maweki 
By the way, is there a way to help you guys on the review process on [extensions.gnome.org/review](https://extensions.gnome.org/review)? If so please let me know. I would like to help out in some way or another :-).

P.S.: I contribute to [Arc-Menu](https://github.com/LinxGem33/Arc-Menu).

**Comment by maweki** [comment]
@lexruee we're basically in need of reviewers, but since we need a base level of trust we're currently only considering gnome foundation members.

**Comment by jwarkentin** [comment]
The new home is https://github.com/gTile/gTile. I just replaced it with a copy that has all of the original history.

**Comment by scherepanov** [comment]
@jwarkentin It appears you forked from vibou/gTile.master
There are changes on scherepanov/gTile that did not make it.
And, last night I made changes that was only in gTile/master.
Will commit all changes when will have a chance

**Comment by jwarkentin** [comment]
@scherepanov I noticed that your fork seems to be missing 6 commits from the official repository here. I only see one new commit added on your repository so I can pull that over.

**Comment by jwarkentin** [comment]
For everyone interested, I have created a new issue to move this discussion over to the new repository at gTile/gTile#1. There's a lot to discuss.

**Comment by scherepanov** [comment]
How it happen... it was a backlog of PR, several people submited identical PR with very small code changes addressing same issue. At same time I had a very big change, doubling code size, and have all those fixes. Last time vibou show up, he start applying all PR, and quickly found they are conflicting. From what I understand he then forced my branch on top, as result was identical to my branch without additional commits. It creates situation when my branch is missing 6 commits but was identical. I start making some changes on my branch again, as it was no access to master. It is a mess. But I would consider my branch as clean cut of latest code. I am looking to copy changed files from my branch over new repo, that should be several small changes, and consider it as a starting point for new code.
